### ISSUE TITLE
Problem with Comparable

### ISSUE THREAD

**Issue by simonv3** [openfarmcc/OpenFarm#603]
```
/Users/simon/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-4.0.2/lib/active_support/callbacks.rb:378: warning: Comparable#== will no more rescue exceptions of #<=> in the next release.
Return nil in #<=> if the comparison is inappropriate or avoid such comparison.
```

@RickCarlino previously opened this here https://github.com/mongoid/mongoid/issues/3928

The folks at mongoid kind of closed it abruptly, but they may be right. Basically there's some places where we're comparing things that just ain't kosher. We should fix that. This *might* be a problem with our test suites!

**Comment by simonv3** [comment]
This is what's causing the warnings: https://redmine.ruby-lang.org/issues/7688

I still don't know whether this is a problem with mongoid. I suspect it is, cause it happens when calling things like .save on a model.

(ex in `mutations/gardens/create_garden.rb`)

**Comment by simonv3** [comment]
Interestingly enough these warnings aren't happening on TravisCI.

**Comment by simonv3** [comment]
Turns out we changed something on TravisCI, and now they are happening there.

**Comment by simonv3** [comment]
This is fixed with upgrading to Rails 4.1.9 and referencing Mongoid 4.0.2 from the gem.
